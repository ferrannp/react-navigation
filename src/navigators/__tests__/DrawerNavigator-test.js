/* @flow */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import DrawerNavigator from '../DrawerNavigator';

const _getScreen = (label: string) => {
  return class extends React.Component {
    static navigationOptions = {
      drawer: () => ({
        label,
      }),
    };

    render() {
      return null;
    }
  }
};

describe('DrawerNavigator', () => {

  const DrawerApp = DrawerNavigator({
    Inbox: {
      screen: _getScreen('Inbox'),
    },
    Drafts: {
      screen: _getScreen('Drafts'),
    },
  });

  it('renders a DrawerNavigator with correct options', () => {
    expect(renderer.create(
      <DrawerApp />,
    )).toMatchSnapshot();
  })

});
