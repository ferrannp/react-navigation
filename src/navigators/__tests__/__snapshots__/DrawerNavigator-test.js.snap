exports[`DrawerNavigator renders a DrawerNavigator with correct options 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "flex": 1,
      }
    } />
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "translateX": -686,
          },
        ],
        "width": 686,
      }
    }>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "paddingTop": 20,
          },
          undefined,
        ]
      }>
      <View
        style={
          Array [
            Object {
              "marginVertical": 4,
            },
            undefined,
          ]
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "backgroundColor": "rgba(0, 0, 0, .04)",
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontWeight": "bold",
                    "margin": 16,
                  },
                  Object {
                    "color": "#2196f3",
                  },
                ]
              }>
              Inbox
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontWeight": "bold",
                    "margin": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, .87)",
                  },
                ]
              }>
              Drafts
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
